### PR TITLE
feat: Add query constraint `wherePolygonContains` to determine whether a point in within a polygon

### PR DESCRIPTION
**Pull request by Paul0Cesar**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

### Issue Description
Create a new constraint to check if Polygon contains the geopoint

Closes: #777 

### Approach

create a new method to add polygon constraint in query

### TODOs before merging
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)

### CODE REVIEW

**Comment by mtrezza** [comment]
Could you add a test?

File: packages/dart/test/parse_query_test.dart [hunk 1]
```diff
@@ -62,5 +64,48 @@ void main() {
...
+              '{"results":[{"objectId":"eT9muOxBTK","createdAt":"2022-07-25T13:46:06.092Z","updatedAt":"2022-07-25T13:46:23.586Z","geometry": {"type": "Polygon","coordinates": [[[84.17724609375,-53.69670647530323],[83.1884765625,-54.61025498157913],[84.814453125,-55.14120964449505],[85.67138671875,-54.40614309031968],[84.17724609375,-53.69670647530323]]]}}]}'));
+
+      ParseResponse response = await queryBuilder.query();
+
+      expect(response.results?.first, isA<ParseObject>());
+
+      ParseObject parseObject = response.results?.first;
+
+      expect(parseObject.objectId, "eT9muOxBTK");
+      expect(parseObject.containsKey("geometry"), true);
```
Comment (mtrezza) [inline packages/dart/test/parse_query_test.dart:95]:
```suggestion
      ParseResponse response = await queryBuilder.query();
      expect(response.results?.first, isA<ParseObject>());
      ParseObject parseObject = response.results?.first;
      expect(parseObject.objectId, "eT9muOxBTK");
      expect(parseObject.containsKey("geometry"), true);
```

File: packages/dart/test/parse_query_test.dart [hunk 2]
```diff
@@ -62,5 +64,45 @@ void main() {
...
+
+      final Uri result = Uri.parse(verify(client.get(
+        captureAny,
+        options: anyNamed("options"),
+        onReceiveProgress: anyNamed("onReceiveProgress"),
+      )).captured.single);
+
+      expect(result.path, '/classes/TEST_SCHEMA');
```
Comment (mtrezza) [inline packages/dart/test/parse_query_test.dart:100]:
```suggestion
      )).captured.single);
      expect(result.path, '/classes/TEST_SCHEMA');
```

**Review by mtrezza — CHANGES_REQUESTED** [review]
Could you take a look at the failing lint test?

**Comment by Paul0Cesar** [comment]
> Could you take a look at the failing lint test?

@mtrezza , sorry for the late answer, I fix the lint error

**Comment by mtrezza** [comment]
Thanks @Paul0Cesar, all tests pass. Let's wait for a review by @parse-community/flutter-sdk-review to merge this.

**Review by mbfakourii — DISMISSED** [review]

**Comment by mtrezza** [comment]
@Paul0Cesar To merge this, could you add the change log entry and bump the version by a minor version number? It seems this is only related to the dart package, so the change log and version number only needs to be changed there.

**Comment by Paul0Cesar** [comment]
@mtrezza , sorry for the late answer, I merge the master branch, fix the constant SINGLE QUERY not using _SINGLE_QUERY, and change the version

**Review by mbfakourii — CHANGES_REQUESTED** [review]
The test must be changed

**Comment by Paul0Cesar** [comment]
@mbfakourii, I change the code formatting, can you check again?

**Review by mbfakourii — DISMISSED** [review]
Thanks for the changes looks good

File: packages/dart/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,9 @@
+## [3.1.11](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-3.1.10...dart-3.1.11) (2023-01-18)
+
+### Features
+
+* Additional query constraint:`wherePolygonContains`([#777](https://github.com/parse-community/Parse-SDK-Flutter/issues/777))
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:5]:
```suggestion
* Add query constraint `wherePolygonContains` to determine whether a point in within a polygon ([#777](https://github.com/parse-community/Parse-SDK-Flutter/issues/777))
```

File: packages/dart/CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,9 @@
+## [3.1.11](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-3.1.10...dart-3.1.11) (2023-01-18)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]:
```suggestion
## [3.1.11](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-3.1.10...dart-3.1.11) (2023-01-20)
```

**Review by mtrezza — DISMISSED** [review]
Minor changelog

**Review by mtrezza — CHANGES_REQUESTED** [review]
Lint fails, could you take a look?

**Comment by Paul0Cesar** [comment]
@mtrezza , done, can you check:?  :rocket:

File: packages/dart/test/parse_query_test.dart [hunk 5]
```diff
@@ -327,5 +327,80 @@ void main() {
...
+
+      // assert
+      expect(response.results?.first, isA<ParseObject>());
+
+      expect(parseObject.objectId, "eT9muOxBTK");
+      expect(parseObject.containsKey("geometry"), true);
+
+      expect(result.path, '/classes/TEST_SCHEMA');
+
+      expect(result.query, expectedQuery.query);
```
Comment (mtrezza) [inline packages/dart/test/parse_query_test.dart:403]:
```suggestion
      expect(response.results?.first, isA<ParseObject>());
      expect(parseObject.objectId, "eT9muOxBTK");
      expect(parseObject.containsKey("geometry"), true);
      expect(result.path, '/classes/TEST_SCHEMA');
      expect(result.query, expectedQuery.query);
```

File: packages/dart/test/parse_query_test.dart [hunk 6]
```diff
@@ -327,5 +327,80 @@ void main() {
...
+        any,
+        options: anyNamed("options"),
+        onReceiveProgress: anyNamed("onReceiveProgress"),
+      )).thenAnswer((_) async => ParseNetworkResponse(
+          statusCode: 200, data: jsonEncode(desiredOutput)));
+
+      // act
+      ParseResponse response = await queryBuilder.query();
+
+      ParseObject parseObject = response.results?.first;
```
Comment (mtrezza) [inline packages/dart/test/parse_query_test.dart:373]:
```suggestion
      ParseResponse response = await queryBuilder.query();
      ParseObject parseObject = response.results?.first;
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: packages/dart/CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,9 @@
+## [3.1.11](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-3.1.10...dart-3.1.11) (2023-01-20)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]:
```suggestion
## [3.1.11](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-3.1.10...dart-3.1.11) (2023-01-21)
```

**Review by mtrezza — APPROVED** [review]
Looks good

**Pull request merged**